packages/web3-core/src/web3_context.ts

```diff
@@ -250,22 +253,85 @@ export class Web3Context<
 		Object.assign(this, _pluginObject);
 	}
 
+	/**
+	 * @returns Returns the current provider
+	 *
+	 * ```ts
+	 * const web3 = new Web3Context("http://localhost:8545");
+	 * console.log(web3.provider);
+	 * > HttpProvider {
+	 * 	clientUrl: 'http://localhost:8545',
+	 * 	httpProviderOptions: undefined
+	 *  }
+	 * ```
+	 */
+
 	public get provider(): SupportedProviders<API> | string | undefined {
 		return this.requestManager.provider;
 	}
 
+	/**
+	 * Will set the current provider.
+	 *
+	 * Accepted providers are of type {@link SupportedProviders}
+	 * ```ts
+	 *  const web3Context = new web3ContextContext("http://localhost:8545");
+	 * web3Context.provider = "ws://localhost:8545";
+	 * console.log(web3Context.provider);
+	 * > WebSocketProvider {
+	 * _eventEmitter: EventEmitter {
+	 * _events: [Object: null prototype] {},
+	 * _eventsCount: 0,
+	 * ...
+	 * }
+	 * ```
+	 */
+
 	public set provider(provider: SupportedProviders<API> | string | undefined) {
 		this.requestManager.setProvider(provider);
 	}
 
+	/**
+	 * @returns Returns the current provider
+	 *
+	 * ```ts
+	 * const web3Context = new Web3Context("http://localhost:8545");
+	 * console.log(web3Context.provider);
+	 * > HttpProvider {
+	 * 	clientUrl: 'http://localhost:8545',
+	 * 	httpProviderOptions: undefined
+	 *  }
+	 * ```
+	 */
```

Comment:
I think it is good to tell the user that this method just does the same as `provider`.